iroh-util/src/lib.rs

```diff
@@ -52,14 +53,19 @@ pub async fn block_until_sigint() {
 
 /// Returns the path to the user's iroh config directory.
 ///
-/// The returned value depends on the operating system and is either a `Some`, containing a value from the following table, or a `None`.
+/// If the `IROH_CONFIG_DIR` environment variable is set it will be used unconditionally.
+/// Otherwise the returned value depends on the operating system according to the following
+/// table.
 ///
 /// | Platform | Value                                 | Example                          |
 /// | -------- | ------------------------------------- | -------------------------------- |
 /// | Linux    | `$XDG_CONFIG_HOME` or `$HOME`/.config/iroh | /home/alice/.config/iroh              |
 /// | macOS    | `$HOME`/Library/Application Support/iroh   | /Users/Alice/Library/Application Support/iroh |
 /// | Windows  | `{FOLDERID_RoamingAppData}`/iroh           | C:\Users\Alice\AppData\Roaming\iroh   |
 pub fn iroh_config_root() -> Result<PathBuf> {
+    if let Some(val) = env::var_os("IROH_CONFIG_DIR") {
+        return Ok(PathBuf::from(val));
```

Comment:
what happens when val isn't a valid filepath? Should this func check & error out?